src/tests/fuchsia.md

```diff
@@ -0,0 +1,162 @@
+
+Fuchsia developers use `fx` to run builds and perform other development tasks.
+This tool is located in `.jiri_root/bin` of the Fuchsia checkout; you may need
+to add this to your `$PATH` for some workflows.
+
+There are a few `fx` subcommands that are relevant, including:
+
+* `fx set` accepts build arguments, writes them to `out/default/args.gn`, and runs GN.
+* `fx build` builds the Fuchsia project using Ninja. It will automatically pick
+  up changes to build arguments and rerun GN. By default it builds everything,
+  but it also accepts target paths to build specific targets (see below).
+* `fx clippy` runs Clippy on specific Rust targets (or all of them). We use this
+  in the Rust CI build to avoid running codegen on most Rust targets. Underneath
+  it invokes Ninja, just like `fx build`. The clippy results are saved in json
+  files inside the build output directory before being printed.
+
+#### Target paths
+
+GN uses paths like the following to identify build targets:
+
+```
+//src/starnix/kernel:starnix_core
+```
+
+The initial `//` means the root of the checkout, and the remaining slashes are
+directory names. The string after `:` is the _target name_ of a target defined
+in the `BUILD.gn` file of that directory.
+
+The target name can be omitted if it is the same as the directory name. In other
+words, `//src/starnix/kernel` is the same as `//src/starnix/kernel:kernel`.
+
+These target paths are used inside `BUILD.gn` files to reference dependencies,
+and can also be used in `fx build`.
+
+#### Modifying compiler flags
+
+You can put custom compiler flags inside a GN `config` that is added to a target.
+As a simple example:
+
+```
+config("everybody_loops") {
+    rustflags = [ "-Zeverybody-loops" ]
+}
+
+rustc_binary("example") {
+    crate_root = "src/bin.rs"
+    # ...existing keys here...
+    configs += [ ":everybody_loops" ]
+}
+```
+
+This will add the flag `-Zeverybody-loops` to rustc for the given target. You can
+also use `public_configs` for a config to be added to every target that depends
+on that target.
+
+If you want to add a flag to every Rust target in the build, you can add
+rustflags to the [`//build/config:compiler`] config or to the OS-specific
+configs referenced in that file. Note that `cflags` and `ldflags` are ignored on
+Rust targets.
+
+#### Running ninja and rustc commands directly
+
+Going down one layer, `fx build` invokes `ninja`, which in turn eventually
+invokes `rustc`. All build actions are run inside the out directory, which is
+usually `out/default` inside the Fuchsia checkout.
+
+You can get ninja to print the actual command it invokes by forcing that command
+to fail, e.g. by adding a syntax error to one of the source files of the target.
+Once you have the command, you can run it from inside the output directory.
+
+After changing the toolchain itself, the build setting `rustc_version_string` in
+`out/default/args.gn` needs to be changed so that `fx build` or `ninja` will
```

Comment:
Do you do this manually? Would it be worthwhile to show here how it looks like in practice, or e.g. could we show a command to do it (à la `fx set rustc_version_string` with `date +%s`)?

Reply:
I normally do it manually in an editor. It's easier than trying to find the right `fx set` command from my history and modifying that.

Reply:
Clarified.